src/coinbase/staking_reward.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  FetchStakingRewardsRequestFormatEnum,
+  StakingReward as StakingRewardModel,
+} from "../client";
+import Decimal from "decimal.js";
+import { Coinbase } from "./coinbase";
+import { Asset } from "./asset";
+import { Amount } from "./types";
+
+/**
+ * A representation of a staking reward earned on a network for a given asset.
+ */
+export class StakingReward {
+  private model: StakingRewardModel;
+  private asset: Asset;
+  private format: FetchStakingRewardsRequestFormatEnum;
+
+  /**
+   * Creates the StakingReward object.
+   *
+   * @param model - The underlying staking reward object.
+   * @param asset - The asset for the staking reward.
+   * @param format - The format to return the rewards in. (usd, native). Defaults to usd.
+   */
+  constructor(
+    model: StakingRewardModel,
+    asset: Asset,
+    format: FetchStakingRewardsRequestFormatEnum,
+  ) {
+    this.model = model;
+    this.asset = asset;
+    this.format = format;
+  }
+
+  /**
+   * Returns a list of StakingRewards for the provided network, asset, and addresses.
+   *
+   * @param networkId - The network ID.
+   * @param assetId - The asset ID.
+   * @param addressIds - The address ID.
+   * @param startTime - The start time.
+   * @param endTime - The end time.
+   * @param format - The format to return the rewards in. (usd, native). Defaults to usd.
+   * @returns The staking rewards.
+   */
+  public static async list(
+    networkId: string,
+    assetId: string,
+    addressIds: Array<string>,
+    startTime: string,
+    endTime: string,
+    format = FetchStakingRewardsRequestFormatEnum.Usd,
+  ): Promise<StakingReward[]> {
+    const stakingRewards: StakingReward[] = [];
+    const queue: string[] = [""];
+
+    while (queue.length > 0) {
+      const page = queue.shift();
+      const request = {
+        network_id: networkId,
+        asset_id: assetId,
```

Comment:
not specific to this line but Asset.primaryDenomination may needed for assetId usage.

```suggestion
        asset_id: Asset.primaryDenomination(assetId),
```